test/UnitTests/MSTest.CoreAdapter.Unit.Tests/Discovery/UnitTestDiscovererTests.cs

```diff
@@ -196,6 +201,29 @@ public void SendTestCasesShouldSendTestCasesWithNaigationDataWhenDeclaredClassFu
         }
 
         [TestMethodV1]
+        public void SendTestCasesShouldSendTestCasesWithNaigationDataWhenDeclaredAssemblyNameIsNonNull()
+        {
+            var source = "DummyAssembly.dll";
+
+            // Setup mocks.
+            this.testablePlatformServiceProvider.MockFileOperations.Setup(fo => fo.CreateNavigationSession(source))
+                .Returns((object)null);
+
+            var test = new UnitTestElement(new TestMethod("M", "C", "A", false));
+            test.TestMethod.DeclaringAssemblyName = "DA";
+
+            var testElements = new List<UnitTestElement> { test };
+
+            this.SetupNavigation(source, test, test.TestMethod.DeclaringClassFullName, test.TestMethod.Name);
+
+            // Act
+            this.unitTestDiscoverer.SendTestCases(source, testElements, this.mockTestCaseDiscoverySink.Object);
+
+            // Assert
+            this.mockTestCaseDiscoverySink.Verify(ds => ds.SendTestCase(It.Is<TestCase>(tc => VerifyTestCase(tc))), Times.Once);
```

Comment:
Strange. It looks like all methods in this class seem to be validating FullyQualifiedName instead of the navigationData(line number and fileName). Changing it to that assertion would fail this test though since the mock setup is done on `DummyAssembly.dll` and we query navigation session on `DA` instead. 
Its perfectly fine if we just fix this one test and I'l take an AI to fix up the rest.